persistence/sql/persister_identity.go

```diff
@@ -281,6 +282,43 @@ func (p *Persister) ListIdentities(ctx context.Context, page, perPage int) ([]id
 	return is, nil
 }
 
+func (p *Persister) ListIdentitiesFiltered(ctx context.Context, values url.Values, page, perPage int) ([]identity.Identity, error) {
+	is := make([]identity.Identity, 0)
+
+	/* #nosec G201 TableName is static */
+	if err := sqlcon.HandleError(p.GetConnection(ctx).Where("identities.nid = ?", corp.ContextualizeNID(ctx, p.nid)).
+		LeftJoin("identity_verifiable_addresses verifiable_addresses", "verifiable_addresses.identity_id=identities.id").
+		LeftJoin("identity_recovery_addresses recovery_addresses", "recovery_addresses.identity_id=identities.id").
+		EagerPreload("VerifiableAddresses", "RecoveryAddresses").
+		Scope(p.buildScope(values)).
+		Paginate(page, perPage).Order("identities.id DESC").
+		All(&is)); err != nil {
+		return nil, err
+	}
+
+	schemaCache := map[string]string{}
+
+	for k := range is {
+		i := &is[k]
```

Comment:
Taking a pointer of the slice element will not make a copy, but use the same underlying memory. Not sure what the intent was here :sweat_smile:

Reply:
it's what I get from the original
https://github.com/ory/kratos/pull/1671/files/751a0e6d80a62eccea872c87e07c57738dfc36ec#diff-e6bdb8b4b6f8cb588c61fbd8410a5adfd4eea9f40eca74740bcac59d63f3f21bR265